[docs][material-ui] Revise and split up "Styled engine" doc (#37774)
mui/material-ui: README.md

yarn add @mui/system @emotion/react @emotion/styled
```

> Or if you want to use styled-components as a styling engine:

Reviewer: ```suggestion
Or if you want to use styled-components as the styling engine:
```

yarn add @mui/material @mui/styled-engine-sc styled-components
```

> Visit our [styled-components guide](https://mui.com/material-ui/guides/styled-components/) for more information on configuration.

Reviewer 2: This page isn't about styled components (https://styled-components.com/)

Reviewer 2: I would expect something like this:

```suggestion
Visit our [CSS engine guide](https://mui.com/material-ui/guides/styled-components/) for more information on configuration.
```

> Visit our [styled-components guide](https://mui.com/material-ui/guides/styled-components/) for more information on configuration.

Reviewer 2: I think that the true location of this guide should be here:

```suggestion
Visit our [styled-components guide](https://mui.com/system/guides/styled-components/) for more information on configuration.
```

But we could say it's beyond the scope of this PR.
